File: View/Components/MatchListViewItem.vb
Imports POFF.Kicker.Model

Namespace Components

    Public Class MatchListViewItem
        Inherits ListViewItem

        Public Sub New(match As Match, Optional matchNo As Integer = 0)
            MyBase.New("", match.Status)
            SubItems.Add($"{matchNo}")
            SubItems.Add(match.Team1.Name)
            SubItems.Add(match.Team2.Name)
            SubItems.Add(match.Result.ToString)

            MatchValue = match
        End Sub

        Private MatchValue As Match
        Public ReadOnly Property Match() As Match
            Get
                Return MatchValue
            End Get
        End Property

        Friend Sub RefreshNumber()
            SubItems(1).Text = $"{Index + 1}"
        End Sub

    End Class

End Namespace

File: POFF.Kicker/Components/MatchListViewItem.cs
using System.Windows.Forms;
using POFF.Kicker.Model;

namespace POFF.Kicker.View.Components;


public class MatchListViewItem : ListViewItem
{

    public MatchListViewItem(Match match, int matchNo = 0) : base("", (int)match.Status)
    {
        SubItems.Add($"{matchNo}");
        SubItems.Add(match.Team1.Name);
        SubItems.Add(match.Team2.Name);
        SubItems.Add(match.Result.ToString());

        MatchValue = match;
    }

    private Match MatchValue;
    public Match Match
    {
        get
        {
            return MatchValue;
        }
    }

    internal void RefreshNumber()
    {
        SubItems[1].Text = $"{Index + 1}";
    }

}